src/peergos/server/social/HttpSocialNetworkServer.java

```diff
@@ -0,0 +1,163 @@
+                        removeFollowRequest(din, dout);
+                        break;
+                    default:
+                        throw new IOException("Unknown method "+ method);
+                }
+
+                dout.flush();
+                dout.close();
+                byte[] b = bout.toByteArray();
+                exchange.sendResponseHeaders(200, b.length);
+                exchange.getResponseBody().write(b);
+            } catch (Exception e) {
+                Throwable cause = e.getCause();
+                if (cause != null)
+                    exchange.getResponseHeaders().set("Trailer", cause.getMessage());
+                else
+                    exchange.getResponseHeaders().set("Trailer", e.getMessage());
+
+                exchange.sendResponseHeaders(400, 0);
+            } finally {
+                exchange.close();
+                long t2 = System.currentTimeMillis();
+                if (LOGGING)
+                    System.out.println("Corenode server handled " + method + " request in: " + (t2 - t1) + " mS");
+            }
+
+        }
+
+        void followRequest(DataInputStream din, DataOutputStream dout) throws Exception
+        {
+            byte[] encodedKey = Serialize.deserializeByteArray(din, PublicSigningKey.MAX_SIZE);
+            PublicKeyHash target = PublicKeyHash.fromCbor(CborObject.fromByteArray(encodedKey));
+            byte[] encodedSharingPublicKey = CoreNodeUtils.deserializeByteArray(din);
+
+            boolean followRequested = social.sendFollowRequest(target, encodedSharingPublicKey).get();
+            dout.writeBoolean(followRequested);
+        }
+        void getFollowRequests(DataInputStream din, DataOutputStream dout) throws Exception
+        {
+            byte[] encodedKey = Serialize.deserializeByteArray(din, PublicSigningKey.MAX_SIZE);
+            PublicKeyHash ownerPublicKey = PublicKeyHash.fromCbor(CborObject.fromByteArray(encodedKey));
+            byte[] res = social.getFollowRequests(ownerPublicKey).get();
+            Serialize.serialize(res, dout);
+        }
+        void removeFollowRequest(DataInputStream din, DataOutputStream dout) throws Exception
+        {
+            byte[] encodedKey = Serialize.deserializeByteArray(din, PublicSigningKey.MAX_SIZE);
+            PublicKeyHash owner = PublicKeyHash.fromCbor(CborObject.fromByteArray(encodedKey));
+            byte[] signedFollowRequest = CoreNodeUtils.deserializeByteArray(din);
+
+            boolean isRemoved = social.removeFollowRequest(owner, signedFollowRequest).get();
+            dout.writeBoolean(isRemoved);
+        }
+    }
+
+    private final HttpServer server;
+    private final InetSocketAddress address;
+    private final SocialHandler ch;
+
+    public HttpSocialNetworkServer(SocialNetwork social, MutablePointers mutable, InetSocketAddress address) throws IOException
+    {
+
+        this.address = address;
+        if (address.getHostName().contains("local"))
```

Comment:
case insensitive?

Reply:
InetSocketAddress already does that for us

Reply:
fixed